internal/handler/mqtthandler/mqtt_handler.go

```diff
@@ -70,23 +75,57 @@ func NewHandler(server, username, password, cafile string) (handler.Handler, err
 	return &h, nil
 }
 
-func newTLSConfig(cafile string) (*tls.Config, error) {
+func newTLSConfig(cafile, certFile, certKeyFile string) (*tls.Config, error) {
+	// Here are three valid options:
```

Comment:
Sorry for commenting on this again, but maybe it is better to remove the hole check at all. E.g. when the user does only provide a tls cert or only a tls key, `tls.LoadX509KeyPair(certFile, certKeyFile)` will also handle this as an error.